tests/expressions/tests.py

```diff
@@ -516,6 +516,18 @@ def test_annotations_within_subquery(self):
             [{'salary': 10, 'total_employees': 2300}, {'salary': 20, 'total_employees': 35}],
         )
 
+    def test_subquery_references_joined_table_twice(self):
+        from .models import Foo, Bar
+
+        bars = Bar.objects.filter(
+            name1=OuterRef('spam1__name'),
+            name2=OuterRef('spam2__name'))
+
+        qs = Foo.objects.annotate(bars=Subquery(bars.values('pk')))
+        qs = qs.filter(bars=True)
+
+        self.assertFalse(qs)
```

Comment:
About Josh's suggestion, we've considered try/except/fail an antipattern because it hides the original exception and thus makes fixing a failure more difficult. There's no problem with a test erroring rather than failing.